### PR TITLE
ci: Update KNOWN_VIOLATIONS list for the linter for functions we don't care are locale-dependent

### PR DESCRIPTION
**Pull request by barton2526**
- Add exceptions for `strftime`
- Remove violations fixed in #2270

### CODE REVIEW

**Comment by barton2526** [comment]
This is connected to #2268

**Review by jamescowens — APPROVED** [review]
ACK

**Comment by jamescowens** [comment]
Waiting on the review to be complete on #2270 before I merge this one.

**Pull request merged**
